fetch-engine.sh

```diff
@@ -20,29 +20,39 @@ fi
 if [ "${AUTOMATIC_ENGINE_MANAGEMENT}" = "True" ]; then
 	echo "OpenRA engine version ${ENGINE_VERSION} is required."
 
-	if [ -d "${ENGINE_DIRECTORY}" ]; then
-		if [ "${CURRENT_ENGINE_VERSION}" != "" ]; then
-			echo "Deleting engine version ${CURRENT_ENGINE_VERSION}."
-		else
-			echo "Deleting existing engine (unknown version)."
+	if command -v git >/dev/null 2>&1; then
+		if [ ! -d "${ENGINE_DIRECTORY}/.git" ]; then
+			rm -rf "${ENGINE_DIRECTORY}"
+			git clone "${AUTOMATIC_ENGINE_SOURCE_GIT}" "${ENGINE_DIRECTORY}"
```

Comment:
Are you able to use `-C` here instead of `--git-dir`?

`-C` allows you to pass just the project root instead of the actual `.git` directory which means, AFAIK, you won't have to also use `--work-tree`.